dashboardwm.py

```diff
@@ -0,0 +1,130 @@
+    return title
+
+
+
+def show_midi_pieces(filtered_dataset, pieces_per_page, current_page, subheader=""):
+    # Calculate the start and end index for the current page
+    start_idx = (current_page - 1) * pieces_per_page
+    end_idx = min(start_idx + pieces_per_page, len(filtered_dataset))
+
+    # Display the records for the selected composer for the current page
+    st.subheader(subheader)
+    cols = st.columns(2)
+
+    for it in range(start_idx, end_idx):
+        record = filtered_dataset[it]
+        piece = MidiPiece.from_huggingface(record)
+        paths = U.piece_av_files(piece)
+
+        # Display the information and media for the MIDI piece in the columns
+        with cols[it % 2]:
+            st.audio(paths["mp3_path"], format="audio/mp3")
+            st.image(paths["pianoroll_path"])
+            st.table(piece.source)
+
+
+def update_page(dataset, option, value, title=""):
+    if option == "composer":
+        filtered_dataset = dataset.filter(lambda example: example["composer"] == value)
+
+    elif option == "title":
+        # search for titles that contain the search string
+        filtered_dataset = dataset.filter(lambda example: value in example["title"])
+
+    return filtered_dataset
+
+
+def main():
+    # Set the layout of the Streamlit page
+    st.set_page_config(layout="wide")
+
+    # select the number of pieces per page
+
+    with st.sidebar:
+        # Create a combo box to select a split (train, test, or validation)
+        pieces_per_page = st.selectbox("Select the number of pieces per page:", [4, 6, 8, 10])
+        selected_split = st.selectbox("Select a split:", ["train", "test", "validation"])
+
+    # Load the dataset for the selected split
+    dataset = load_data(selected_split)
+
+    # Get unique composer names from the dataset
+    composers = get_composers(dataset)
+
+    if not composers:
+        st.warning("No composers found for the selected split.")
+        return
+
+    # Create a combo box to select a composer
+    selected_composer = st.sidebar.selectbox("Select a composer:", composers)
+
+    # Get titles by the selected composer
+    titles = get_titles_by_composer(dataset, selected_composer)
+
+    if not titles:
+        st.warning("No pieces found")
+        return
+
+    # Create a select box to choose from the titles of the selected composer
+    search_by_title = st.sidebar.selectbox("Search by title:", titles)
+
+    # Create a radio button to select the filter option (composer or title)
+    filter_option = st.sidebar.radio("Filter by:", ["composer", "title"])
+
+    if filter_option == "composer" and selected_composer:
+        filtered_dataset = dataset.filter(lambda example: example["composer"] == selected_composer)
+    elif filter_option == "title" and search_by_title:
+        filtered_dataset = dataset.filter(lambda example: str(search_by_title) in example["title"])
+    else:
+        filtered_dataset = dataset
```

Comment:
```suggestion
    filtered_dataset = dataset.filter(lambda example: str(search_by_title) in example["title"])
```

Let's try an interface where there is only one way of filtering: select the composer -> select the title. In that case we don't need this whole if statement :)